src/sentry/receivers/outbox/control.py

```diff
@@ -43,6 +46,22 @@ def process_integration_updates(object_identifier: int, region_name: str, **kwds
 @receiver(process_control_outbox, sender=OutboxCategory.SENTRY_APP_UPDATE)
 def process_sentry_app_updates(object_identifier: int, region_name: str, **kwds: Any):
 
+    # This function should only be used when the sentry app is being deleted.
+    # Currently this receiver is only used for deletion.
+    if options.get("workflow_engine.sentry-app-actions-outbox"):
+        logger.info(
+            "sentry_app_update.update_action_status",
+            extra={
+                "region_name": region_name,
+                "sentry_app_id": object_identifier,
+            },
+        )
+        action_service.update_action_status_for_sentry_app_via_sentry_app_id(
+            region_name=region_name,
+            status=ObjectStatus.DISABLED,
+            sentry_app_id=object_identifier,
+        )
```

Comment:
Depending on how long this RPC takes to complete, this might make us time out 🤔 We can test it for now, but I just realized this may be problematic.